### PR TITLE
Removing SecretStr from RemoteRepository model

### PR DESCRIPTION
**Pull request by ajrpeggio**
SecretStr is not JSON serializable and is throwing exceptions for RemoteRepositories that require a password.

## Description
I've been using this library to create repositories for my Artifactory Instance. I started adding a RemoteRepository for docker and came upon this error that states: "Object of type SecretStr is not JSON serializable"

## To Reproduce

```python
from pyartifactory import Artifactory
from pyartifactory.models import RemoteRepository

url = <url to artifactory instance>
auth = (<username>, <password>)
remote_docker_repo_cfg = {
    "key": "docker-remote",
    "url": "https://registry-1.docker.io/",
    "packageType": "docker",
    "username": <dockerhub username>,
    "password": <dockerhub_password>,
}
rt = Artifactory(url=url, auth=auth)
remote_repo = RemoteRepository(**remote_docker_repo_cfg)
rt.repositories.create_repo(remote_repo)
```

Fixes # https://github.com/anancarv/python-artifactory/issues/74

### CODE REVIEW

**Comment by anancarv** [comment]
Hi @ajrpeggio ,
After running some tests. It appears that you can keep the `SecretStr` from the `RemoteRepository` model. Also, changing it to str would decrease the level of security and our password will be able to be read.
Instead, in the `objects.py` file, you can change the following line:
`self._put(f"api/{self._uri}/{repo_name}", json=repo.dict())` to `self._put(f"api/{self._uri}/{repo_name}", json=repo.json())`

Indead, .json() will be able to serialize the SecretStr param

**Comment by anancarv** [comment]
Hi @ajrpeggio,
After running some tests. It appears that it's better to keep the `SecretStr` type in the RemoteRepository model. Indeed, changing it to `str` will decrease the level of security and your password will be able to be printed.

A better solution is t create a custom encoder in a `utils.py` file for example with the following content:
```
import json
from pydantic import SecretStr
from pydantic.json import pydantic_encoder

def custom_encoder(obj):
      if isinstance(obj, SecretStr):
          return obj.get_secret_value()
      else:
          return pydantic_encoder(obj)
```

Then in the `objects.py` file, in the file 466 you can replace:
```
self._put(f"api/{self._uri}/{repo_name}", json=repo.dict())
```
with
```
data= json.dumps(repo, default=custom_encoder)
self._put(f"api/{self._uri}/{repo_name}", headers={"Content-Type": "application/json"}, data=data)
```

File: pyartifactory/utils.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from pydantic import SecretStr
+from pydantic.json import pydantic_encoder
+
+
+def custom_encoder(obj):
+    if isinstance(obj, SecretStr):
+        return obj.get_secret_value()
+    else:
+        return pydantic_encoder(obj)
```
Comment (anancarv) [inline pyartifactory/utils.py:9]: It seems that you haven't run the `pre-commits`. If you had, you could have seen the following errors:
```
pyartifactory/utils.py:1:0: C0114: Missing module docstring (missing-module-docstring)
pyartifactory/utils.py:5:0: C0116: Missing function or method docstring (missing-function-docstring)
pyartifactory/utils.py:6:4: R1705: Unnecessary "else" after "return" (no-else-return)
```

Can be fixed with:
```
"""
Definition of all utils.
"""

from typing import Any

from pydantic import SecretStr
from pydantic.json import pydantic_encoder


def custom_encoder(obj: Any) -> Any:
    """
    Custom encoder function to be passed to the default argument of json.dumps()
    :param obj: A pydantic object
    :return: An encoded pydantic object
    """
    if isinstance(obj, SecretStr):
        return obj.get_secret_value()
    return pydantic_encoder(obj)
```

File: pyartifactory/objects.py [hunk 2]
```diff
@@ -1,952 +1,961 @@
...
+        """
+        repo_name = repo.key
+        try:
+            self.get_repo(repo_name)
+            logger.error("Repository %s already exists", repo_name)
+            raise RepositoryAlreadyExistsException(
+                f"Repository {repo_name} already exists"
+            )
+        except RepositoryNotFoundException:
+            data= json.dumps(repo, default=custom_encoder)
```
Comment (anancarv) [inline pyartifactory/objects.py:470]: The pre-commits also highlight formatting issue in this line. Indeed `data=` should be `data =`

**Review by anancarv — CHANGES_REQUESTED** [review]

**Comment by anancarv** [comment]
Also, there are conflicts with the master branch. Indeed, a new release has been merged a few days ago. You will also need to fix the conflicts before merging

**Comment by anancarv** [comment]
Hi @ajrpeggio ,
Are you facing troubles with the changes requested ? Would you prefer me to it ?

**Review by anancarv — APPROVED** [review]

**Review by nymous — APPROVED** [review]
Let's go!

**Comment by ajrpeggio** [comment]
Apologies for getting back to this so late. Thank you for fixing these!

**Comment by anancarv** [comment]
No worries. You're welcome !

**Pull request merged**